### PR TITLE
4.0.0. release

### PR DESCRIPTION
**Pull request by mabar**
options.data works and options.includePaths throws error on invalid data type given.

New version released due to BC break.

### CODE REVIEW

**Comment by mabar** [comment]
Related to #599

**Comment by mabar** [comment]
@dlmanning All tests passed, can you merge it?

**Comment by stevelacy** [comment]
Just wondering, why is this a major change? It does not alter the deps, nor break the api - it just modifies the conditionals

**Comment by mabar** [comment]
It may be a break for developers with misconfigured options.includePaths and options.data. To this time it was just ignored, so it is a BC break.
But importance of major version change is not so big in this case. I can change it to 3.2.0 if required by maintainers.

**Comment by hsingh23** [comment]
Why isn't this merged?

**Comment by sielay** [comment]
@mabar have you published it elsewhere already?

**Comment by mabar** [comment]
@sielay No, I am using directly my fork. But don't use it, it can be deleted anytime. I am going to switch to webpack as more and more gulp integrations are not maintained.

**Comment by sielay** [comment]
@mabar sad, especially webpack is config hell. thanks

**Comment by xzyfer** [comment]
I am not convinced about this change. Please see my comments on https://github.com/dlmanning/gulp-sass/issues/599#issuecomment-372105344

File: index.js [hunk 1]
```diff
@@ -50,6 +55,9 @@ var gulpSass = function gulpSass(options, sync) {
       if (typeof opts.includePaths === 'string') {
         opts.includePaths = [opts.includePaths];
       }
+      else {
+        throw new Error('options.includePaths expected to be a string, ' + typeof opts.includePaths + ' given.');
```
Comment (xzyfer) [inline index.js]: I believe the existing behavior is correct. `opts.includePaths` can be either a string or an array of strings.
The type of check coerces a string into an array of one string.
This else branch is the default case. This error will break gulp-sass for most users.

File: index.js [hunk 2]
```diff
@@ -35,8 +35,13 @@ var gulpSass = function gulpSass(options, sync) {
 
 
     opts = clonedeep(options || {});
-    opts.data = file.contents.toString();
-
+    if (opts.data === void 0) {
+      opts.data = file.contents.toString();
+    }
+    else {
+      opts.data = opts.data + file.contents.toString();
```
Comment (xzyfer) [inline index.js]: Please see my comments on https://github.com/dlmanning/gulp-sass/issues/599#issuecomment-372105425 and #302 for why I don't think this change is correct.

**Pull request closed without merging**